### PR TITLE
Tooltips: Make tooltips non interactive by default

### PR DESCRIPTION
**Pull request by torkelo**
Our current tooltips can often create a really bad user experience because they do not automatically hide when you hover out from the trigger. If you want to interact with something behind the tooltip you can't. 

This has been raised in an issue can't find it though. 

Video describing/showing the biggest problem 
https://user-images.githubusercontent.com/10999/152971612-ce4e904d-f247-4f82-a4c2-72bb4d3faa2d.mp4

* Updates our Tooltip to not use PopoverController (think we should deprecate this) 
* Uses usePopperTooltip as it has all the features we need (simple abstractions ontop of usePopper) 
* Show tooltip on focus not just hover? (good idea? option?) 
* Adds an interactive property that is default false that controls if a tooltip can be hovered onto (and interacted with, ex click a link)

There is no critical breaking change (should not be), but a breaking behavior here. So existing Tooltip uses that have links inside need to add the property `interactive` to get the same behavior. 

# Release notice breaking change

The Tooltip component provided by `@grafana/ui` is no longer automatically interactive (that is you can hover onto it and click a link or select text). It will from now on by default close automatically when you mouse out from the trigger element. To make tooltips behave like before set the new `interactive` property to true.

### CODE REVIEW

File: packages/grafana-ui/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -1,61 +1,189 @@
...
-export type PopoverContent = string | React.ReactElement<any> | ((props: PopoverContentProps) => JSX.Element);
+export const Tooltip = React.memo(({ children, theme, interactive, show, placement, content }: TooltipProps) => {
+  const { getArrowProps, getTooltipProps, setTooltipRef, setTriggerRef, visible, update } = usePopperTooltip({
+    visible: show,
+    placement: placement,
+    interactive: interactive,
+    delayHide: interactive ? 100 : 0,
+    delayShow: 50,
+    // Focus is new, old tooltip did not show on focus, we could make this a parameter/option?
+    trigger: ['hover', 'focus'],
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Tooltip/Tooltip.tsx:29]: From the a11y POV I think we should keep the focus to open the tooltips automatically. Can't think of any situation when provided tooltip would not be shown in keyboard navigation scenario.

So IMO the behavior presented by this PR is desired one.

File: public/app/features/dashboard/dashgrid/PanelHeader/PanelHeaderCorner.tsx [hunk 2]
```diff
@@ -86,7 +86,7 @@ export class PanelHeaderCorner extends Component<Props> {
     const ariaLabel = selectors.components.Panels.Panel.headerCornerInfo(infoMode.toLowerCase());
 
     return (
-      <Tooltip content={content} placement="top-start" theme={theme}>
+      <Tooltip content={content} placement="top-start" theme={theme} interactive>
```
Comment (dprokop) [inline public/app/features/dashboard/dashgrid/PanelHeader/PanelHeaderCorner.tsx:89]: Interactive works as expected 👍

**Review by dprokop — APPROVED** [review]
LGTM. Think this is a nice improvement, was frustrating to me as well. Also - I like the subtle design update to the tooltips, they now look more refined.

**Comment by leeoniya** [comment]
cc @nmarrs is doing a lot of tooltip work currently. also possibly related: https://github.com/grafana/grafana/issues/45032

**Comment by aocenas** [comment]
Copying this from slack:

personally would be for keeping the interactivity. Especially in the case @torkelo shown in the video there is lots of info in those tooltips and I hate when I have elements in the page with docs for example from which I cannot copy paste stuff.

Like if it is just to give a name to an icon button, that is probably no issue, but we tend to put lots of docs into those tooltips lots of times it seems.

For this particular issue, I would rather have those info buttons to open the tooltip on click instead on hover for example which I think would solve this and be probably more UX friendly (like if you are poweruser you don't need those at all so it does not make sense to even flash them when moving your mouse around)

**Review by axelavargas — COMMENTED** [review]
Hey Torkel 👋🏾 , here are some thoughts after taking a look:

- Making non-interactive by default: I think adding this change will potentially break existing  use cases that we can miss during this PR, for example when we use tooltip JSON ([today in Grafana play](https://play.grafana.org/d/0000000121/table-json-tooltip-closes-when-extended-34553?orgId=1))

And this is how will look with this change: 
tooltipJson

Of course, this just opens a different discussion, I think this problem is the consequence of using Tooltips and Popovers as the same thing, **maybe this is a good opportunity of making a clear difference?**

Tooltip: When we want to show a short text, it will be visible when hovering or focusing over a word or icon, should not receive focus.

Popover: Show a longer text, or when you want to have a link to an external web page. It is shown when clicking over a word or icon. The window remains visible until they click again outside of the popover.

An idea 💡 :
We could create 2 new components (TooltipV2 and Popover) and use them correctly? 


> Show tooltip on focus not just hover? (good idea? option?) 

Based on [a11y guidelines is recommended we do that](https://dequeuniversity.com/library/aria/tooltip)

**Comment by torkelo** [comment]
@axelavargas I have fixed the JSONView tooltip, good catch, added the new property ìnteractive` to this tooltip so you can hover onto it like before. 

Yea, think it might be a good idea to create a separate Popover component (we have one but it's not great) for larger Tooltips. But can explore that in another PR

**Pull request merged**